### PR TITLE
 add (user/enemy)(hit/miss) events to weaponspecial

### PR DESCRIPTION
**Pull request by jay20162016**
Useful for some weapon specials, like poison

### CODE REVIEW

File: source/game/Game.gd [hunk 1]
```diff
@@ -247,16 +247,26 @@ func _on_attack_selected(combatChoices: Dictionary, target: Unit) -> void:
 		if attacker_strikes > 0:
 			combatChoices['offense'].execute_each_turn(temp_current, target, true)
 			if randi() % 100 > target.get_defense():
+				combatChoices["offense"].execute_user_hit(temp_current, target, true)
```
Comment (iwek7) [inline source/game/Game.gd:250]: you could consider moving `offense` and `defense` strings to variables since they are reused multiple times

**Comment by iwek7** [comment]
I'd consider if really we want to add this code to master as it is doing nothing (just unimplemented interface calls) - it is hard to check if it it working (only possible by debugging) and as such it might become buggy in future and we wouldn't even know

**Comment by jay20162016** [comment]
We would probably need this for some weapon specials, such as poison and slow.

**Comment by iwek7** [comment]
I reckon, but `probably` is the keyword here. When implementing weapon specials it might turn out that other set of methods is needed or that implementer wants to do this different way.

**Comment by jay20162016** [comment]
We would like everything to mostly be similar to the original wesnoth, in which case units are poisoned/slowed when the attack hits, which is one of the events in this case

**Comment by Byteron** [comment]
with the experimental branch close to being ready to merge, this is obsolete, hence closing it.

**Pull request closed without merging**